python/BioSimSpace/FreeEnergy/_relative.py

```diff
@@ -531,6 +531,13 @@ def _analyse_somd(work_dir=None):
         if proc.returncode != 0:
             raise _AnalysisError("SOMD free-energy analysis failed!")
 
+        #rerun without subsampling if the subsampling has resulted in less than 50 samples
+        with open("%s/mbar.txt" % work_dir) as file:
+            for line in file:
+                if "#WARNING SUBSAMPLING ENERGIES RESULTED IN LESS THAN 50 SAMPLES" in line:
+                    command = "%s mbar -i %s/lambda_*/simfile.dat -o %s/mbar.txt --overlap" % (_analyse_freenrg, work_dir, work_dir)
+                    proc = _subprocess.run(command, shell=True, stdout=_subprocess.PIPE, stderr=_subprocess.PIPE)
+        
```

Comment:
I'd duplicate the check of the process return code from above in case anything else happened to go wrong re-running the command without subsampling. At present, it would just proceed directly to the analysis, perhaps using the original output file if the second command failed. Also put a `break` statement after the return code check so that you skip looping over the rest of the file.

I'm not sure if it's 100% necessasry, but it might also be worth informing the user that their sampling is limited when this fallback is triggered?

Reply:
Thanks, @annamherz. This looks good.

Reply:
Okay, thank you! I can't really think of a good place to include it other than as a print statement, as the function otherwise only returns the data and overlap for that one leg?